WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -242,6 +245,8 @@ public void run() {
                             ShortcutsNavigator.ACTION_OPEN_SHORTCUT), this, getSelectedSite());
                 } else if (openRequestedPage) {
                     handleOpenPageIntent(getIntent());
+                } else if (ActivityUtils.isDeepLinking(getIntent())) {
```

Comment:
That's great, thanks! Because of this I researched how [Android suggests](https://developer.android.com/training/implementing-navigation/temporal#SynthesizeBackStack) we open the Activity *AND* have the back stack and they suggest exactly what you said:

```
Ordinarily, the system incrementally builds the back stack as the user navigates from one activity to the next. However, when the user enters your app with a deep link that starts the activity in its own task, it's necessary for you to synthesize a new back stack because the activity is running in a new task without any back stack at all.

For example, when a notification takes the user to an activity deep in your app hierarchy, you should add activities into your task's back stack so that pressing Back navigates up the app hierarchy instead of exiting the app. This pattern is described further in the Navigation design guide.
```